### PR TITLE
[TT-5186] revert max_content_length, tests

### PR DESCRIPTION
**Pull request by titpetric**
## Description

Revert changes for `max_content_length`.
Update config dos for `max_request_body_length`.
Improve tests

## How This Has Been Tested

```
#!/bin/bash
function request {
	curl -X -vvv -XPOST -H "Content-Type: text/plain" $@
}

request -d"abcde-0000" http://localhost:8080/httpbin-test/headers
echo

request -d"abcde-00001" http://localhost:8080/httpbin-test/headers
echo

request -d"abcde-00001" -H"Transfer-Encoding:chunked" http://localhost:8080/httpbin-test/headers
echo

request -d"abcde-0000" -H"Transfer-Encoding:chunked" http://localhost:8080/httpbin-test/headers
echo
```

The main issue has been how to get an undefined length body from outside go tests. Using Transfer-Encoding headers omits content-length. The max_request_body_size has been set to 10 on the gateway.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5052/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5058865526)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5052/merge
Commit: 0e768acd7fe455c403feff291157ecf39b67c7d6 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5059015378)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5052/merge
Commit: 1cc1a4e6e33f819f362339b0aee762583db95113 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5059411887)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5052/merge
Commit: ca76a42fba1eb0c7f612bc28ca5849b681e19296 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5060007163)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5052/merge
Commit: 7a9a0d581eb58159049c7680636e54010face10b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5059930749)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5052/merge
Commit: e73cfc375c848d3143f0d82d776a33c3255ae1d4 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5062811359)

File: internal/httputil/request.go [hunk 1]
```diff
@@ -2,14 +2,17 @@ package httputil
 
 import "net/http"
 
-// IsTransferEncodingChunked checks that a http.Request is not
-// chunked. Requests with this transfer encoding do not include
-// a valid Content-Length header.
-func IsTransferEncodingChunked(req *http.Request) bool {
+// TransferEncoding gets the header value from the request.
```
Comment (jeffy-mathew) [inline internal/httputil/request.go:5]: are we using these funcs anywhere other than tests?

Comment (titpetric) [reply internal/httputil/request.go:5]: no, internal

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5052/merge
Commit: db3691d980b37202b4c751746332aadc6430a768 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5069306065)

**Pull request merged**